frontend/src/component/NavigationBar.tsx

```diff
@@ -0,0 +1,126 @@
+import React from "react";
+import {discordLoginUri, getUserAvatar} from "../constants";
+import {isLoggedIn, getUser, logout, getCurrentUserPermissions} from "../service/auth";
+
+import {
+    AppBar,
+    Avatar,
+    Box,
+    Button,
+    Container, FormControl, FormControlLabel,
+    IconButton,
+    Menu,
+    MenuItem, Radio, RadioGroup,
+    Toolbar,
+    Tooltip,
+    Typography, useColorScheme
+} from "@mui/material";
+import {useNavigate} from "react-router-dom";
+import {ProtectedRouteButton} from "./ProtectedRouteButton";
+
+export default function NavigationBar() {
+    const { mode, setMode } = useColorScheme();
+    const navigate = useNavigate();
+
+    const userLoggedIn = isLoggedIn();
+    const userPermissions = getCurrentUserPermissions();
+
+    const [anchorElUser, setAnchorElUser] = React.useState<null | HTMLElement>(null);
+    const handleOpenUserMenu = (event: React.MouseEvent<HTMLElement>) => {
+        setAnchorElUser(event.currentTarget);
+    };
+    const handleCloseUserMenu = () => {
+        setAnchorElUser(null);
+    };
+
+    const logoutAndRedirect = () => {
+        logout();
+        navigate("/");
+    };
+
+    let userElement = <Button href={discordLoginUri}>Login</Button>;
+    if (userLoggedIn) {
+        const user = getUser();
+        if (user === null) {
+            throw new Error("User cannot possibly be null");
+        }
```

Comment:
```suggestion
        const user = getUser()!;
```
Throwing an error for the sole purpose of throwing is a bad practice, this either should be handled, or discard it if we're sure it's ok.